### ISSUE TITLE
Unintuitive behavior with `autoPinEdges(toSuperviewMarginsExcludingEdge: .right)`

### ISSUE THREAD

**Issue by michaelkirk** [PureLayout/PureLayout#209]
The common pattern of pinning three of four edges to the superview is nicely encapsulated in:
`autoPinEdgesToSuperviewEdgesWithInsets:(ALEdgeInsets)insets excludingEdge:(ALEdge)edge`

But the implementation only makes sense if you are using Leading/Trailing. 

It doesn't make sense to mix notions of Left/Right with Leading/Trailing when positioning a single view, but (e.g.) if you specify "except left", it will pin "trailing". Instead it should pin "right".

For example, if I'm pinning all edges except the "Right" edge, then it should be the "Left" edge that is pinned, not the "Leading" edge.

Conversely if I'm pinning all edges except the "Left" edge, then it should the "Right" edge that is pinned, not the "Trailing" edge.

Current implementation for reference:
```
- (PL__NSArray_of(NSLayoutConstraint *) *)autoPinEdgesToSuperviewEdgesWithInsets:(ALEdgeInsets)insets excludingEdge:(ALEdge)edge
{
    PL__NSMutableArray_of(NSLayoutConstraint *) *constraints = [NSMutableArray new];
    if (edge != ALEdgeTop) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeTop withInset:insets.top]];
    }
    if (edge != ALEdgeLeading && edge != ALEdgeLeft) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeLeading withInset:insets.left]];
    }
    if (edge != ALEdgeBottom) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeBottom withInset:insets.bottom]];
    }
    if (edge != ALEdgeTrailing && edge != ALEdgeRight) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeTrailing withInset:insets.right]];
    }
    return constraints;
}
```
Proposed changes (I can open a PR if it seems directionally agreeable):
```
- (PL__NSArray_of(NSLayoutConstraint *) *)autoPinEdgesToSuperviewEdgesWithInsets:(ALEdgeInsets)insets excludingEdge:(ALEdge)edge
{
    PL__NSMutableArray_of(NSLayoutConstraint *) *constraints = [NSMutableArray new];
    if (edge != ALEdgeTop) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeTop withInset:insets.top]];
    }
    if (edge != ALEdgeBottom) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeBottom withInset:insets.bottom]];
    }

   // Are we dealing with leading/trailing? Or left/right?
    if (edge == ALEdgeLeading) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeTrailing withInset:insets.right]];
    }
    if (edge == ALEdgeTrailing) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeLeading withInset:insets.left]];
    }
    if (edge == ALEdgeLeft) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeRight withInset:insets.right]];
    }
    if (edge == ALEdgeRight) {
        [constraints addObject:[self autoPinEdgeToSuperviewEdge:ALEdgeLeft withInset:insets.left]];
    }

    return constraints;
}
```

**Comment by toohotz** [comment]
Agreed that in a RTL situation, `leading` should not be assumed to be equal to `left` as they differ in their meanings.  This also made me look into the work that I did with the safe areas and also noticed that the same would have to be done there as well.  I see no reason as to why a PR can't be made for this although it would be nice to get the safe area stuff merged in prior to and having these changes made in one full sweep instead of having to come back and do it for those additional methods.  (Trying to get a hold of @mickeyreiss to sort out the PRs on the repo as things are still in an ongoing process)

**Comment by toohotz** [comment]
So just wanted to get back to you on this one.  Given that generally when constraints are created, it is normally assumed to be using leading / trailing constraints where needed.  I will update the README but per previous discussions, it would be a better choice to individually create each constraint if you so desire to use left / right as opposed to leading / trailing because for most users of the API that would be their desired choice of constraints to be used.

Though reading over this quote from Apple's auto layout guide I am curious as to when using leading / trailing does RTL not handle accordingly as they stated that it is supposed to.

> The attributes leading and trailing are the same as left and right for left-to-right languages such as English, but in a right-to-left environment such as Hebrew or Arabic, leading and trailing are the same as right and left. When you create constraints, leading and trailing are the default values. You should usually use leading and trailing to make sure your interface is laid out appropriately in all languages, unless you’re making constraints that should remain the same regardless of language (such as the order of master and detail panes in a split view).

[Anatomy of Constraints](https://developer.apple.com/library/content/documentation/UserExperience/Conceptual/AutolayoutPG/AnatomyofaConstraint.html#//apple_ref/doc/uid/TP40010853-CH9-SW1)

It is discouraged to use Left / Right per their documentation.
> Avoid using Left and Right attributes. Use Leading and Trailing instead. This allows the layout to adapt to the view’s reading direction.

The choice is of course yours how you intend to use the API but given Leading / Trailing is the ideal way of doing so, the PureLayout API will continue to prefer the choice of Leading / Trailing over Left / Right.

**Comment by michaelkirk-signal** [comment]
Thanks for you work on this.

@toohotz  - I think what you are saying is that in general we should be using leading/trailing instead of left/right.

I agree.

Though *very* occasionally there are things that you want on the left (or right) regardless of language direction. e.g. in messaging apps it's conventional for the "send" button to be on the right regardless of language direction.

**Comment by toohotz** [comment]
@michaelkirk-signal Agreed with the occasion to always have the send button for example on the right side regardless of the language direction and because of that, I will update the API's docs to denote that if you desire to use left / right that creating the edge constraints would be better done individually since the `-[autoPinEdges:]` assumes for you which may not be your intent all the time.
